src/ingestion/utils.py

```diff
@@ -25,21 +33,174 @@ def convert_html_to_markdown(html_file: str) -> str:
     """
-    ---- PLACEHOLDER VERSION ----
-    ---- TO BE MODIFIED ----
     Chunks the input text into smaller segments of specified size.
 
     Args:
         text (str): The text to be chunked.
-        chunk_size (int): The maximum number of words per chunk. Default is 300.
+        chunking_mode (str): The type of chunking to use. Options: 'recursive', 'markdown_specific', or 'html_specific'. Default is 'recursive'.
+        max_chunk_size (int): The maximum number of token per chunk. Default is 200.
+        min_chunk_size (int): The minimum number of token per chunk. Default is 50.
+        chunk_overlap (int): The number of overlapping tokens between consecutive chunks. Default is 0.
+        headers_level (int): Header level to use for splitting in 'markdown_specific' and 'html_specific' modes. Default is 6.
+        include_headers (bool): If True, includes headers as part of chunk splitting in 'markdown_specific' and 'html_specific' modes. Default is True.
 
     Returns:
         List[str]: A list of text chunks.
     """
-    words = text.split()
-    chunks = [
-        " ".join(words[i : i + chunk_size]) for i in range(0, len(words), chunk_size)
-    ]
+    splits = None
+
+    if chunking_mode == "markdown_specific":
+        # Markdown-based text splitting
+        headers_to_split_on = [ ("#"*(i+1), f"Headers {i+1}") for i in range(headers_level)]
+        
+        if include_headers:
+            markdown_splitter = MarkdownHeaderTextSplitter(headers_to_split_on, strip_headers=False)
+        else:
+            markdown_splitter = MarkdownHeaderTextSplitter(headers_to_split_on)
+        
+        try:
+            splits = markdown_splitter.split_text(text)
+        except ValueError as e:
+            logger.error(f"Markdown-based text splitting is not possible, proceeding with no-format splitting")
+        
+    elif chunking_mode == "html_specific":
+        # HTML-based text splitting
+        headers_to_split_on = [ (f"h{i+1}", f"Headers {i+1}") for i in range(headers_level)]
+
+        if include_headers:
+            html_splitter = HTMLSectionSplitter(headers_to_split_on)
+        else:
+            html_splitter = HTMLHeaderTextSplitter(headers_to_split_on)
+
+        try:
+            splits = html_splitter.split_text(text)
+        except ValueError as e:
+            logger.error(f"HTML-based text splitting is not possible, proceeding with no-format splitting")
+
+    # Recursive text splitting
+    text_splitter = RecursiveCharacterTextSplitter.from_tiktoken_encoder(
+        encoding_name="cl100k_base",
+        separators=["  \n", " \n", ".\n", ". ", ".", "\n", " ", ""],
+        chunk_size=max_chunk_size,
+        chunk_overlap=chunk_overlap
+        )
+
+    if not splits:
+        splits = text_splitter.create_documents([text])
+
+    splits = text_splitter.split_documents(splits)
```

Comment:
pensavi di mettere questa linea in un else, oppure vuoi applicare create_documents e split_documents in sequenza anche con lo stesso splitter (caso no-splits)?

Reply:
L'idea è di effettuare lo splitting ricorsivo di base nel caso in cui non si riesca ad effettuare quello per formato specifico, oppure nel caso in cui non venga richiesto un chunking per formato specifico nella chiamata alla funzione. Quindi non pensavo di aggiungere un else, il comportamento dovrebbe essere quello atteso in questo modo.